### PR TITLE
[process-agent] Fix process CPU usage percentages being reported as negative values

### PR DESCRIPTION
**Pull request by kkhor-datadog**
### What does this PR do?

Prevents the process agent from reporting negative CPU usage percentage value for processes.

### Motivation

Broadly speaking, CPU usage percentages on both *nix and Windows systems is computed by dividing the amount of time the process has spent executing by the total CPU execution time, and in rare scenarios, the operating system may report negative values for these quantities, resulting in negative CPU usage percentage values being computed. 

For Linux systems, CPU time counters used to determine process execution time may potentially be decremented as described in https://0xstubs.org/debugging-a-flaky-cpu-steal-time-counter-on-a-paravirtualized-xen-guest/,  and hence lead to the process agent collecting negative process execution times. On Windows systems, the process agent approximates the total CPU execution time using the elapsed system time, which can become negative if the system time is manually reset, and thus cause negative CPU usage percentage values.

This patch adds a check to the process agent's CPU percentage computation logic in the process check on both *nix and Windows systems to handle cases where negative values are reported, and return a CPU percentage value of 0 instead. A similar check in the CPU percentage computation logic already exists for the container check.

### Describe how to test/QA your changes

- Set up the core agent and process agent on a Windows machine and verify that process metrics are being reported correctly for the host in Live Processes.
- **With the process agent running**, go to the Windows date and time settings (accessible by clicking on the date and time in the lower right -> Date and time settings) and disable automatic clock synchronization ("Set time automatically").
- Reset the system time to 2 minutes in the past.
- Monitor process CPU usage metrics being reported in Live Processes and ensure that negative CPU usage values are not reported for the host.
- Optionally, do some basic testing on a Linux machine and verify that CPU usage values look correct for the host in Live Processes

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/process/checks/process_common_test.go [hunk 1]
```diff
@@ -108,11 +108,21 @@ func TestPercentCalculation(t *testing.T) {
...
 	if runtime.GOOS != "windows" {
+		// on *nix systems, CPU utilization is multiplied by number of cores to emulate top
 		assert.True(t, floatEquals(calculatePct(100, 200, 2), 100))
 		assert.True(t, floatEquals(calculatePct(0.04, 8.08, 8), 3.960396))
 		assert.True(t, floatEquals(calculatePct(1.09, 8.08, 8), 107.920792))
+	} else {
+		// on Windows, CPU utilization is not multiplied by number of cores
+		assert.True(t, floatEquals(calculatePct(100, 200, 2), 50))
+		assert.True(t, floatEquals(calculatePct(0.04, 8.08, 8), 0.4950495))
+		assert.True(t, floatEquals(calculatePct(1.09, 8.08, 8), 13.490099))
```
Comment (kkhor-datadog) [inline pkg/process/checks/process_common_test.go:125]: This functionality has not changed in this PR - the tests are added just to validate existing behavior

**Review by hestonhoffman — APPROVED** [review]
Docs 👍

**Review by just-chillin — APPROVED** [review]
Amazing!!!

**Review by xornivore — APPROVED** [review]
:shipit:

File: pkg/process/checks/process_nix.go [hunk 2]
```diff
@@ -76,5 +77,9 @@ func calculatePct(deltaProc, deltaTime, numCPU float64) float32 {
 	}
 
 	// In order to emulate top we multiply utilization by # of CPUs so a busy loop would be 100%.
-	return float32(overalPct * numCPU)
+	pct := overalPct * numCPU
```
Comment (just-chillin) [inline pkg/process/checks/process_nix.go:80]: Do you mind summarizing the "motivation" section here? It would be nice to have this captured in code

**Pull request merged**